tests/transforms/test_get_unitary_matrix.py

```diff
@@ -0,0 +1,215 @@
+# Copyright 2018-2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from functools import reduce
+import pytest
+import numpy as np
+from gate_data import I, X, H, S, CNOT
+import pennylane as qml
+
+from pennylane.transforms.get_unitary_matrix import get_unitary_matrix
+
+# test non-parametric single qubit gates
+nonparam_1qubit_op_list = [qml.PauliX, qml.PauliY, qml.PauliZ, qml.Hadamard, qml.S, qml.T, qml.SX]
+
+
+@pytest.mark.parametrize("op", nonparam_1qubit_op_list)
+@pytest.mark.parametrize("wire", [0, 1, 2])
+def test_get_unitary_matrix_nonparam_1qubit_ops(op, wire):
+
+    wires = [0, 1, 2]
+
+    def testcircuit(wire):
+        op(wires=wire)
+
+    get_matrix = get_unitary_matrix(testcircuit, wires)
+    matrix = get_matrix(wire)
+
+    if wire == 0:
+        expected_matrix = np.kron(op(wires=wire).matrix, np.eye(4))
+    if wire == 1:
+        expected_matrix = np.kron(np.eye(2), np.kron(op(wires=wire).matrix, np.eye(2)))
+    if wire == 2:
+        expected_matrix = np.kron(np.eye(4), op(wires=wire).matrix)
+
+    assert np.allclose(matrix, expected_matrix)
+
+
+# Test a circuit containing multiple gates
+def test_get_unitary_matrix_multiple_ops():
+
+    wires = ["a", "b", "c"]
```

Comment:
This is great, it's always a good idea to test with a set of wires that aren't just integers :100: